eval/eval.go

```diff
@@ -81,6 +138,13 @@ func (s *State) evalAssignment(right object.Object, node *ast.InfixExpression) o
 	}
 	switch node.Left.Value().Type() {
 	case token.DOT, token.LBRACKET:
+		nodeType := node.Type()
+		if isCompound(nodeType) {
+			opToEval := nodeType - (token.SUMASSIGN - token.PLUS)
```

Comment:
maybe isCompound can be/have 

func CompoundOp(type) (op, bool)

so you'd do
```go
if opToEval, isCompound = CompoundOp(nodeType); isCoumpoud {
```
as it's there in 2 places that semi hacky substract